modules/runner/src/main/java/org/apache/ignite/internal/configuration/NodeBootstrapConfiguration.java

```diff
@@ -0,0 +1,118 @@
+ */
+@FunctionalInterface
+public interface NodeBootstrapConfiguration {
+    /**
+     * Path to node configuration file.
+     *
+     * @return Path to node configuration file in HOCON format.
+     */
+    Path configPath();
+
+    /**
+     * Simple config file provider.
+     *
+     * @param configPath Path to node bootstrap configuration.
+     * @return Simple implementation with provided configuration file.
+     */
+    static NodeBootstrapConfiguration directFile(Path configPath) {
+        return () -> configPath;
+    }
+
+    /**
+     * Return node bootstrap configuration with content from {@param is}.
+     *
+     * @param is Configuration content.
+     * @param workDir Dir for configuration file location.
+     * @return Node bootstrap configuration with lazy config file creation.
+     */
+    static NodeBootstrapConfiguration inputStream(@Nullable InputStream is, Path workDir) {
+        return new NodeBootstrapConfiguration() {
+
+            private final AtomicReference<Path> config = new AtomicReference<>();
+
+            @Override
+            public Path configPath() {
+                if (config.compareAndSet(null, createEmptyConfig(workDir))) {
+                    if (is != null) {
+                        try {
+                            Files.write(config.get(), is.readAllBytes(), StandardOpenOption.DSYNC);
+                        } catch (IOException e) {
+                            throw new NodeConfigReadException("Failed to read config input stream.", e);
+                        }
+                    }
+                }
+                return config.get();
+            }
+        };
+    }
+
+    /**
+     * Return node bootstrap configuration with content from {@param plainConf}.
+     *
+     * @param plainConf Configuration content.
+     * @param workDir Dir for configuration file location.
+     * @return Node bootstrap configuration with lazy config file creation.
+     */
+    static NodeBootstrapConfiguration string(@Nullable @Language("HOCON") String plainConf, Path workDir) {
+        return inputStream(plainConf != null
+                ? new ByteArrayInputStream(plainConf.getBytes(StandardCharsets.UTF_8))
+                : null,
+                workDir);
+    }
+
+    /**
+     * Empty config provider.
+     *
+     * @param workDir Configuration file location.
+     * @return Node bootstrap configuration provider to empty config.
+     */
+    static NodeBootstrapConfiguration empty(Path workDir) {
+        return () -> createEmptyConfig(workDir);
+    }
+
+    private static Path createEmptyConfig(Path workDir) {
+        try {
+            Path config = workDir.resolve("ignite-config.conf");
+            File file = config.toFile();
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+            return config;
+        } catch (IOException e) {
+            throw new NodeConfigCreateException("Failed to create temp conf file.", e);
```

Comment:
Why is it a temp file? Name suggests otherwise. And I guess no one deletes it afterwards